vta/apps/tsim_example/tests/python/test_tsim.py

```diff
@@ -0,0 +1,38 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import tvm
+import numpy as np
+from tsim.load import load_driver, load_tsim
+
+def test_tsim(i):
+    rmin = 1 # min vector size of 1
+    rmax = 64
+    n = np.random.randint(rmin, rmax)
+    ctx = tvm.cpu(0)
+    a = tvm.nd.array(np.random.randint(rmax, size=n).astype("uint64"), ctx)
+    b = tvm.nd.array(np.zeros(n).astype("uint64"), ctx)
+    tsim = load_tsim()
+    f = load_driver()
+    f(tsim, a, b)
```

Comment:
could we also show here how to extract statistics from the simulation assuming it's not too complicated? things like cycle count etc?

Reply:
Currently, it prints the number of cycles taken by an entire execution `[DONE] at cycle:0000000000000343`

This is printed by default from the Host DPI module writing in Verilog, since it is the module in charge of handling the accelerator. People could add other event counters to measure more fine-grained events either through this DPI module or as part of the accelerator. I believe this is one of those things that will depend on the design itself maybe?